### PR TITLE
Fix #116 add xlsx module

### LINKED ISSUE #116: create xlsx module
Need to generate a spreadsheet with logical relationships in blocks so can refer to cells out of order. Use xlsxwriter to generate the spreadsheet, but need to have multiple passes to establish links , formats, defaults (e.g. rounding), etc.

### PR DESCRIPTION
**Pull request by robnagler**
- Require test?
   - yes
- Security implications?
    - none
- UI changes reviewed?
    - none
- Backwards compatible?
    - new module
- Require doc changes?
    - yes
- Add [DesignHints](https://github.com/radiasoft/pykern/wiki/DesignHints)?
    - no

### CODE REVIEW

**Comment by robnagler** [comment]
Test failed due to xml output issue. Need to figure out how to sort the attributes.

File: pykern/xlsx.py [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+            fmt = fmt.replace('.0', x)
+            return fmt
+
+        for k in (
+            'font',
+            'border',
+        ):
+            if k in cell.defaults:
+                self.update(self._MAP[cell.defaults[k]])
+        for x in cell.get('fmt', '').split(' '):
```
Comment (mkeilman) [inline pykern/xlsx.py:481]: Is this guaranteed to be a single space, or is the usual regexp (```/\s+/```) better?

File: pykern/xlsx.py [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+
+        Args:
+            title (str): debug label for the table
+            defaults (PKDict): default values, e.g. round_digits
+        """
+        return self._child(self.tables, _Table, kwargs)
+
+    def width(self, col, width):
+        if col not in self._col_widths or self._col_widths[col] <= width:
+            pkdp([self, col, width])
```
Comment (mkeilman) [inline pykern/xlsx.py:588]: Stray ```pkdp```

**Review by mkeilman — CHANGES_REQUESTED** [review]
Just a few comments

File: tests/xlsx_data/2.in/case.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+PATH = 'case2.xlsx'
+w = pykern.xlsx.Workbook(path=PATH)
+s = w.sheet(title='s1')
+t = s.table(title='t1', defaults=PKDict(round_digits=0, num_fmt='currency'))
+t.header(
+    name='Name',
+    count='Count',
+)
+x = list()
```
[the remark this answers is not in the document]
Comment (robnagler) [reply tests/xlsx_data/2.in/case.py]: Fix fbaa131

File: pykern/xlsx.py (hunk 1 above)
Comment (robnagler) [reply pykern/xlsx.py]: Fix fbaa131

File: pykern/xlsx.py (hunk 2 above)
Comment (robnagler) [reply pykern/xlsx.py]: Fix fbaa131

**Review by mkeilman — APPROVED** [review]
Looks good

File: tests/xlsx_data/2.in/case.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+for i in range(1, 5):
+    t.row(
+        name=f'c{i}',
+        count=t.cell(i, link='count'),
+    )
+t.footer(
+    name='Total',
+    count=t.cell(['*', 'count'], link='prod')
+)
```
[the remark this answers is not in the document]
Comment (robnagler) [reply tests/xlsx_data/2.in/case.py:26]: @mkeilman c9de1dc contains a fix to the problem you saw. The string values are stored in sharedStrings.xml and that's what changed when you changed "R" to something else so it wasn't being tested before c9de1dc.

**Comment by robnagler** [comment]
@e-carlin please comment on this PR. I just wanted it merged since it is a new module so it is unlikely to break anything to merge.

**Pull request merged**